src/app/code/community/Divante/VueStorefrontIndexer/Model/Resource/Catalog/Category.php

```diff
@@ -48,14 +48,25 @@ public function getCategories($storeId = 1, array $categoryIds = [], $fromId = 0
         $rootCategoryId = Mage::app()->getStore($storeId)->getRootCategoryId();
         $rootCategory = Mage::getModel('catalog/category')->load($rootCategoryId);
 
-        $select = $this->connection->select()->from(['e' => $this->coreResource->getTableName('catalog/category')]);
+        $isFlatEnabled = Mage::helper('catalog/category_flat')->isEnabled();
+
+        if($isFlatEnabled) {
+            $tableName = sprintf('catalog_category_flat_store_%d', $storeId);
+            $select = $this->connection->select()->from(['e' => $tableName ]);
+            $select->where('e.include_in_menu = ?', 1);
+        } else {
+            $select = $this->connection->select()->from(['e' => $this->coreResource->getTableName('catalog/category')]);
+        }
```

Comment:
I agree with Agata in here

Reply:
I understand. Its true that flat data might not be up-to-date, but we did this when there was still no async indexing (at least not that I was aware of) and I assumed that it would be a general requierement to show only "include_in_menu" categories. Further works on joining attributes tables may be required then.